Rubberduck.VBEEditor/Extensions/ComponentTypeExtensions.cs

```diff
@@ -30,10 +37,40 @@ public static string FileExtension(this ComponentType componentType)
                 case ComponentType.Document:
                     // documents should technically be a ".cls", but we need to be able to tell them apart.
                     return DocClassExtension;
-                case ComponentType.ActiveXDesigner:
+                case ComponentType.PropPage:
+                    return PropertyPageExtension;
+                case ComponentType.UserControl:
+                    return UserControlExtension;
+                case ComponentType.DocObject:
+                    return DocObjectExtension;
                 default:
                     return string.Empty;
             }
         }
+
+        public static IDictionary<string, ComponentType> ComponentTypeForExtension(VBEKind vbeKind)
+        {
+            return vbeKind == VBEKind.Hosted
+                ? VBAComponentTypeForExtension
+                : VB6ComponentTypeForExtension;
+        }
+
+        private static readonly IDictionary<string, ComponentType> VBAComponentTypeForExtension = new Dictionary<string, ComponentType>
+        {
+            [StandardExtension] = ComponentType.StandardModule,
+            [ClassExtension] = ComponentType.ClassModule,
+            [FormExtension] = ComponentType.UserForm,
+            [DocClassExtension] = ComponentType.Document
+        };
+
+        private static readonly IDictionary<string, ComponentType> VB6ComponentTypeForExtension = new Dictionary<string, ComponentType>
+        {
+            [StandardExtension] = ComponentType.StandardModule,
+            [ClassExtension] = ComponentType.ClassModule,
+            [FormExtension] = ComponentType.VBForm,
```

Comment:
`.frm` also maps to MDIForm, not sure if that matters...

Reply:
I'll do a follow up PR to correct this.